File: tooling/templates/template.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <errno.h>

/* --solution-comment-- */

int main(int argc, char *argv[]) {
    const char *filename = argc > 1? argv[1] : "input"; 
    FILE *fp = fopen(filename, "r");
    if (fp == NULL) {
        fprintf(
            stderr,
            "Failed to open file \"%s\": %s.\n",
            filename,
            strerror(errno)
        );
        return EXIT_FAILURE;  
    }
}

File: tooling/templates/template.py
#!/usr/bin/env python3

# --solution-comment--

from collections.abc import Iterable
from sys import argv


def main(input_lines: Iterable[str]):
    ...


if __name__ == "__main__":
    with open(argv[1] if argv[1:] else "input", "r") as f:
        print(main(f))
